### ISSUE TITLE
Dialog sometimes hidden

### ISSUE THREAD

**Issue by fleckdalm** [vaadin/vaadin-date-picker#133]
When using the app-header element of the new app-layout (https://github.com/PolymerLabs/app-layout) parts of the dialog are oftentimes hidden behind the app-header. It should be positioned in a way so that it resides always on top of all other elements or in the middle of the screen. 

![vaadin-datepicker](https://cloud.githubusercontent.com/assets/12670319/13917164/662cf500-ef5f-11e5-88c2-b8cba431a2cb.png)
Don't wonder why the bottom bar is gone I removed it as it isn't needed in my app. Maybe make this configuarable?

Apart from that, this is the best date-picker component i found so far, thanks!

**Comment by tomivirkki** [comment]
Glad you like the component :) I was trying to reproduce the issue by adding vaadin-date-pickers around the sections of app-layout in this example project https://github.com/frankiefu/summit but wasn't able to spot the issue there.
Could you provide more info on how we could reproduce it. A code example would be very helpful.

There's no property for hiding the footer on the dropdown but you can apply a mixin to hide it like so:
```
<style is="custom-style">
  vaadin-date-picker {
    --vaadin-date-picker-footer: {
      display: none;
    }
  }
</style>
```

**Comment by fleckdalm** [comment]
Thanks!

Unfortunately I can't provide real code as I'm working on a project for my company. 
I spent an hour trying to debug the problem but wasn't able to track it down completely and unfortunately I'm unable to invest more time in this now.

It seems somehow to be related to iron-swipeable-pages (or in general residing in a container with position: absolute). 

Here is a vague strcuture of my code:
`
 <app-drawer-layout id="paperDrawerPanel">
                <app-drawer id="drawer" swipe-open="true" class="drawerPanel" fixed>
                    
                    <my-app-menu menu-route="{{menuRoute}}" route="{{route}}"></my-app-menu>
                </app-drawer>

                <app-header-layout>
                    
                    <app-header id="mainHeader" class="mainAppHeader" shadow condenses fixed>
                        
                        <app-toolbar class="flexRow flexSpaceBetween flexVerticalCenter titleToolbar">
                          test
                        </app-toolbar>
                    </app-header>


                    
                    <div class="content">
                     <iron-swipeable-pages threshold="0.1" transition-duration="150" selected="{{selectedPage}}" no-cycle="true">
 <vaadin-date-picker id="datePicker" label="{{label}}" value="{{date}}"></vaadin-date-picker>
</iron-swipeable-pages>
            <section>
                    </div>
                </app-header-layout>
            </app-drawer-layout>
`

**Comment by tomivirkki** [comment]
Seems that iron-swipeable-pages uses "will-change: transform" rule for it's direct children. It creates a stacking context which together with app-header-layout contentContainer's "overflow-y: hidden;" causes the dropdown to clip.

The problem therefor concerns all components using iron-dropdown/fixed elements.

One workaround for the issue is to add "will-change: auto" to all iron-swipeable-pages' children (in the example's case I used &lt;section style="will-change: auto"&gt;).

Side note: We're planning on replacing iron-dropdown with a custom dropdown in the future (#143) which might affect this also.

Second side note: I noticed you've styled the date picker to match your app's theme. We just made a change to the themability api and removed all the custom css variables for now.
If you used the custom css properties for styling, you'll need to switch to the mixins or default paper-styles css properties (example: https://github.com/vaadin/vaadin-date-picker/blob/master/demo/index.html#L114) while updating to alpha6/beta (once released). Just a heads up :)

**Comment by fleckdalm** [comment]
Thanks for the investigation!
I tried a lot of things but never would have thought that this might be causing it. How did you find out?

To disable will-change is unfortunately no option for me as it greatly decreases performance on mobile. Could you think of anything else I could do?

Thanks as well for the second side note, good to be informed :)

**Comment by tomivirkki** [comment]
Just had a hunch that this might relate to stacking contexts :)
Would it be an option for you to force "will-change: auto" only while the dropdown is open? We might need to add public events for this later but for now you could listen to 'iron-overlay-opened' and 'iron-overlay-closed' which are fired internally by iron-dropdown.

Unfortunately we can't automatically do this for all ancestor elements since stacking context can be created in a few different ways and resetting some of them might affect the page appearance/layout.

**Comment by peterblazejewicz** [comment]
@tomivirkki 
I've hit the same problem. If I use `vaading-date-picker` directly on rendered content section:
```html
<section data-route="reports">
    <div class="layout horizontal around-justified">
        <vaadin-date-picker id="start" label="Start"></vaadin-date-picker>
        <vaadin-date-picker id="end" label="End"></vaadin-date-picker>
    </div>
```
It seems quite OK (except of some styling quirks). But when I nest component within `paper-card` content, the layout of component is broken and it renders in the incorrect depth (below the application bar):
```html
<paper-material>
    <paper-card heading="{{teamReport.Team}}" elevation="1" class="grid-chart-card">
        <div class="card-content">
            <h2>{{teamReport.Description}}</h2>
            <div class="layout horizontal around-justified">
                <vaadin-date-picker id="start" label="Start"></vaadin-date-picker>
                <vaadin-date-picker id="end" label="End"></vaadin-date-picker>
            </div>
        </div>
    </paper-card>
```
(ignore ad-hoc styling in the screenshots):
![productivity reports - google chrome 2016-04-20 08 56 13](https://cloud.githubusercontent.com/assets/14539/14666214/8b192e34-06d8-11e6-97df-4d709e2c156b.png)
![productivity reports - google chrome 2016-04-20 09 04 34](https://cloud.githubusercontent.com/assets/14539/14666216/8b19d87a-06d8-11e6-818b-fa5b5c6220af.png)
![productivity reports - google chrome 2016-04-20 09 03 56](https://cloud.githubusercontent.com/assets/14539/14666215/8b19301e-06d8-11e6-903f-1189f5e95976.png)

Thanks!

**Comment by peterblazejewicz** [comment]
> Seems that iron-swipeable-pages uses "will-change: transform" rule for it's direct children

@tomivirkki 
Exactly, I've been using `will-change` in custom paper-card implementation. As soon as I've removed that property the component works reasonably fine :+1:

**Comment by fleckdalm** [comment]
I think this could be related: https://github.com/PolymerElements/iron-overlay-behavior/pull/155 ;)

**Comment by marcozanghi** [comment]
I am using modal with this:
```
    var body = document.querySelector('body');
    Polymer.dom(body).appendChild(this.$.dialogId);
    this.$.dialogId.open()`
```

**Comment by jouni** [comment]
We are most likely going to wait for iron-overlay and this PR to get merged (https://github.com/PolymerLabs/iron-overlay/pull/3), and fix this issue for vaadin-date-picker 2.0.

**Comment by web-padawan** [comment]
Fixed by #482 and released as part of `v3.0.0-alpha1`
